homeassistant/components/decora_wifi/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Will write later."""
+
+import logging
+
+from decora_wifi import DecoraWiFiSession
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN, HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BASE_SCHEMA = vol.Schema(
+    {
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+class DecoreWifiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Decora Wifi Integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_data: dict[str, str]) -> FlowResult:
+        """Handle importting decora wifi config from configuration.yaml."""
+
+        async_create_issue(
```

Comment:
Okay so the thing with issues and importing: since I believe ADR-21 we need to raise different issues when different stuff occurs.

So when the import succeeds we show the general issue that the import succeeded and they should remove it from YAML. But we should raise issues for every bad flow, for example, what if the config was already imported? What if the credentials are incorrect? What if the login doesn't raise unauthorized but cannot connect?

We should show our best effort in letting the user know why the import failed.

Reply:
Ok let's see if I understand this correctly - 
1. if user has the integration config in yaml, we would want to show a "deprecate_yaml". Issue so they can remove it since that is deprecated. This would be true regardless of good or bad credentials?
2. If auth failed (incorrect credentials or otherwise) - you also want that to create a sep. issue? Can you share an example of that? This would be in addition to raising config error which would show in UI that integration is not ready?

Reply:
https://github.com/home-assistant/core/pull/104962 this is something I worked on yesterday we can use as an example. (It's not reviewed yet so it's subject to change, but this would be the general direction).

You can see I raise different config issues when stuff happens. (Now that i think of it I miss the issue when import works).